src/jit/morph.cpp

```diff
@@ -12422,6 +12422,33 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 }
             }
 
+            //
+            // Here we look for the following transformation
+            //
+            //                        EQ/NE                 EQ/NE
+            //                        /  \                  /   \
+            //                    (U)MOD  CNS 0    ->     AND   CNS 0
+            //                     /   \                 /   \
+            //                    x    CNS              x    CNS-1
+            //
+            // X % C == 0 can be transformed to X & (C-1) == 0 if C is a power of two
+            if (op2->IsIntegralConst(0))
+            {
+                op1 = tree->gtOp.gtOp1;
+                if ((op1->gtOper == GT_MOD) || (op1->gtOper == GT_UMOD))
+                {
+                    if (varTypeIsIntegralOrI(op1->gtOp.gtOp1->gtType) && (op1->gtOp.gtOp2->IsIntegralConst()))
+                    {
+                        ssize_t divider = op1->gtOp.gtOp2->AsIntCon()->IconValue();
+                        if ((divider > 1) && (isPow2(divider)))
```

Comment:
Is the `divider > 1` check really needed? Also, you have unnecessary paranthesese around `isPow2`